### PR TITLE
[Blade] Discord Event Integration

### PR DESCRIPTION
**Pull request by DVidal1205**
# Why
Discord Events are crucial to club event attendance, since they format good announcements and notifications to boost engagement. This PR adds Discord integrations to the Event Management dashboard to create the event automatically.
# What
- Adds Discord REST interactions on the event related procedures (createEvent, updateEvent, and deleteEvent).
- Adds some QOL form validations concerning date and time.
- Added an endtime and starttime field to the form and database, required by Discord metadata.
# Test Plan

https://github.com/user-attachments/assets/122fccaa-4a6b-44c3-805d-d87a7adbede4

### CODE REVIEW

File: packages/api/src/routers/event.ts [hunk 1]
```diff
@@ -1,41 +1,146 @@
...
+        const endIsoTimestamp = endDatetime.toISOString();
+
+        const response = (await discord.post(
+          Routes.guildScheduledEvents(KNIGHTHACKS_GUILD_ID),
+          {
+            body: {
+              description: input.description,
+              name:
+                "[" +
+                input.tag.toUpperCase().replace(" ", "-") +
```
Comment (Lewin-B) [inline packages/api/src/routers/event.ts:38]: Might be better to define this as a variable for readability.

File: packages/api/src/routers/event.ts [hunk 2]
```diff
@@ -1,41 +1,146 @@
...
+                "] " +
+                input.name,
+              privacy_level: 2,
```
Comment (Lewin-B) [inline packages/api/src/routers/event.ts:41]: I would define this as a const

File: packages/api/src/routers/event.ts [hunk 3]
```diff
@@ -1,41 +1,146 @@
...
+              scheduled_start_time: startIsoTimestamp,
+              scheduled_end_time: endIsoTimestamp,
+              entity_type: 3,
```
Comment (Lewin-B) [inline packages/api/src/routers/event.ts:44]: I would also define this as a const

File: packages/api/src/routers/event.ts [hunk 4]
```diff
@@ -1,41 +1,146 @@
...
+        await discord.patch(
+          Routes.guildScheduledEvent(KNIGHTHACKS_GUILD_ID, input.discordId),
+          {
+            body: {
+              description: input.description,
+              name:
+                "[" +
+                input.tag.toUpperCase().replace(" ", "-") +
+                "] " +
+                input.name,
```
Comment (Lewin-B) [inline packages/api/src/routers/event.ts:96]: Refer to the create event procedure comments

File: packages/consts/src/knight-hacks.ts [hunk 5]
```diff
@@ -42,9 +42,9 @@ export const EVENT_POINTS: Record<EventTag, number> = {
 
 export const KNIGHTHACKS_MEMBERSHIP_PRICE = 2500;
 
-export const DISCORD_ADMIN_ROLE_ID = "1319413082258411652";
+export const DISCORD_ADMIN_ROLE_ID = "1321955700540309645";
 
-export const KNIGHTHACKS_GUILD_ID = "486628710443778071";
+export const KNIGHTHACKS_GUILD_ID = "1151877367434850364";
```
Comment (Lewin-B) [inline packages/consts/src/knight-hacks.ts:47]: If these were switched for testing then they should be switched back

**Review by Lewin-B — CHANGES_REQUESTED** [review]
Nice Job! should be good to merge just had some concerns with the guild and role ids also some left some nits.

File: packages/consts/src/knight-hacks.ts (hunk 5 above)
Comment (DVidal1205) [reply packages/consts/src/knight-hacks.ts:47]: whoops, big fumble ty

**Review by Lewin-B — APPROVED** [review]
lgtm

**Pull request merged**